src/Linkify/Linkify.js

```diff
@@ -8,9 +8,21 @@ const Container = styled.span`
   }
 `;
 
-const REGEX = /(\b(https?|ftp|file):\/\/[-A-Z0-9+&@#\/%?=~_|!:,.;]*[-A-Z0-9+&@#\/%=~_|])/gi;
+const escapeHtmlChars = (unsafe = '') =>
+  unsafe
+    .replace(/&/g, '&amp;')
+    .replace(/</g, '&lt;')
+    .replace(/>/g, '&gt;')
+    .replace(/"/g, '&quot;')
+    .replace(/'/g, '&#039;');
 
-const parse = string => string.replace(REGEX, url => `<a target="_blank" href="${url}">${url}</a>`);
+const REGEX = /(\b(https?|ftp|file):\/\/[-A-Z0-9+&@#/%?=~_|!:,.;]*[-A-Z0-9+&@#/%=~_|])/gi;
+
+const parse = string =>
+  escapeHtmlChars(string).replace(
+    REGEX,
+    url => `<a target="_blank" rel="noopener noreferrer" href="${url}">${url}</a>`
+  );
 
 const Linkify = props => (
   <Container style={props.style || {}} dangerouslySetInnerHTML={{ __html: parse(props.children) }} />
```

Comment:
i think you can do this without `dangerouslySetInnerHTML` if you iterate through regex matches and return a `[React.Component]`. see https://github.com/tasti/react-linkify/blob/master/src/components/Linkify.jsx#L75 (could also use that library directly, up to you)

Reply:
While I think using a library like this would be ideal, our use case for this component usually resolves parsing strings as opposed to React elements. So, for example, we currently use the component like this:
```jsx
<Linkify>
  {message.body} //=> string
</Linkify>
```
as opposed to like this. 
```jsx
<Linkify>
  <span>Hello this a tag</span>
  <span>https://link.com</span>
</Linkify>
```
While I definitely do agree about with the general notion of making code as general and extensible as possible, I think in this case I would hold off on addressing edge cases until and if we get to them. 

What do you think? @choochootrain 😄

Reply:
discussed offline-but-actually-online-because-slack - i think if we defer html escaping to react then we can be p confident that future bugs will have a lot more eyeballs on them :)